client/lib/catch-js-errors/index.js

```diff
@@ -11,24 +11,27 @@ export default class ErrorLogger {
 
 		if ( ! window.onerror ) {
 			TraceKit.report.subscribe( errorReport => {
+				const error = {
+					message: errorReport.message,
+					url: document.location.href
+				};
+
 				if ( Array.isArray( errorReport.stack ) ) {
-					errorReport.stack.forEach( report => Object.keys( report ).forEach( key => {
+					const trace = errorReport.stack.slice( 0, 10 );
+					trace.forEach( report => Object.keys( report ).forEach( key => {
 						if ( key === 'context' && report[ key ] ) {
-							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 512 );
+							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 256 );
```

Comment:
Okay, so `256` chars of the error report is enough to deduce something interesting out of it? I mean, we can reduce this all we want but we also want something to get from it. :)